2017-05-23 00:52:51,588 INFO  [main] reduce.TestFetcher (TestFetcher.java:setup(99)) - >>>> testCopyFromHostWithRetry
2017-05-23 00:52:53,418 INFO  [main] reduce.TestFetcher (TestFetcher.java:teardown(124)) - <<<< testCopyFromHostWithRetry
2017-05-23 00:52:53,421 INFO  [main] reduce.TestFetcher (TestFetcher.java:setup(99)) - >>>> testCopyFromHostIncompatibleShuffleVersion
2017-05-23 00:52:53,508 WARN  [main] reduce.Fetcher (Fetcher.java:openShuffleUrl(287)) - Failed to connect to localhost with 2 map outputs
java.io.IOException: Incompatible shuffle response version
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.verifyConnection(Fetcher.java:465)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.setupConnectionsWithRetry(Fetcher.java:407)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.openShuffleUrl(Fetcher.java:271)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:335)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostIncompatibleShuffleVersion(TestFetcher.java:250)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
2017-05-23 00:52:53,517 WARN  [main] reduce.Fetcher (Fetcher.java:openShuffleUrl(287)) - Failed to connect to localhost with 2 map outputs
java.io.IOException: Incompatible shuffle response version
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.verifyConnection(Fetcher.java:465)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.setupConnectionsWithRetry(Fetcher.java:407)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.openShuffleUrl(Fetcher.java:271)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:335)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostIncompatibleShuffleVersion(TestFetcher.java:250)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
2017-05-23 00:52:53,523 WARN  [main] reduce.Fetcher (Fetcher.java:openShuffleUrl(287)) - Failed to connect to localhost with 2 map outputs
java.io.IOException: Incompatible shuffle response version
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.verifyConnection(Fetcher.java:465)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.setupConnectionsWithRetry(Fetcher.java:407)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.openShuffleUrl(Fetcher.java:271)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:335)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostIncompatibleShuffleVersion(TestFetcher.java:250)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
2017-05-23 00:52:53,533 INFO  [main] reduce.TestFetcher (TestFetcher.java:teardown(124)) - <<<< testCopyFromHostIncompatibleShuffleVersion
2017-05-23 00:52:53,534 INFO  [main] reduce.TestFetcher (TestFetcher.java:setup(99)) - >>>> testCopyFromHostIncompatibleShuffleVersionWithRetry
2017-05-23 00:52:53,580 WARN  [main] reduce.Fetcher (Fetcher.java:openShuffleUrl(287)) - Failed to connect to localhost with 2 map outputs
java.io.IOException: Incompatible shuffle response version
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.verifyConnection(Fetcher.java:465)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.setupConnectionsWithRetry(Fetcher.java:407)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.openShuffleUrl(Fetcher.java:271)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:335)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostIncompatibleShuffleVersionWithRetry(TestFetcher.java:282)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
2017-05-23 00:52:53,588 WARN  [main] reduce.Fetcher (Fetcher.java:openShuffleUrl(287)) - Failed to connect to localhost with 2 map outputs
java.io.IOException: Incompatible shuffle response version
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.verifyConnection(Fetcher.java:465)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.setupConnectionsWithRetry(Fetcher.java:407)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.openShuffleUrl(Fetcher.java:271)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:335)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostIncompatibleShuffleVersionWithRetry(TestFetcher.java:282)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
2017-05-23 00:52:53,594 WARN  [main] reduce.Fetcher (Fetcher.java:openShuffleUrl(287)) - Failed to connect to localhost with 2 map outputs
java.io.IOException: Incompatible shuffle response version
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.verifyConnection(Fetcher.java:465)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.setupConnectionsWithRetry(Fetcher.java:407)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.openShuffleUrl(Fetcher.java:271)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:335)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostIncompatibleShuffleVersionWithRetry(TestFetcher.java:282)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
2017-05-23 00:52:53,601 INFO  [main] reduce.TestFetcher (TestFetcher.java:teardown(124)) - <<<< testCopyFromHostIncompatibleShuffleVersionWithRetry
2017-05-23 00:52:53,602 INFO  [main] reduce.TestFetcher (TestFetcher.java:setup(99)) - >>>> testCopyFromHostConnectionRejected
2017-05-23 00:52:53,641 WARN  [main] reduce.Fetcher (Fetcher.java:openShuffleUrl(278)) - Connection rejected by the host localhost. Will retry later.
2017-05-23 00:52:53,690 INFO  [main] reduce.TestFetcher (TestFetcher.java:teardown(124)) - <<<< testCopyFromHostConnectionRejected
2017-05-23 00:52:53,692 INFO  [main] reduce.TestFetcher (TestFetcher.java:setup(99)) - >>>> testCopyFromHostWithRetryUnreserve
2017-05-23 00:52:53,746 INFO  [Thread-10] reduce.Fetcher (Fetcher.java:copyMapOutput(558)) - fetcher#9 about to shuffle output of map null decomp: 10 len: 10 to null
2017-05-23 00:52:53,747 WARN  [Thread-10] reduce.Fetcher (Fetcher.java:copyMapOutput(564)) - Failed to shuffle for fetcher#9
java.io.IOException: forced error
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:561)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:348)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostWithRetryUnreserve(TestFetcher.java:717)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$StatementThread.run(FailOnTimeout.java:74)
2017-05-23 00:52:53,748 WARN  [Thread-10] reduce.Fetcher (Fetcher.java:checkTimeoutOrRetry(622)) - Shuffle output from localhost failed, retry it.
java.io.IOException: java.io.IOException: forced error
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:565)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:348)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostWithRetryUnreserve(TestFetcher.java:717)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$StatementThread.run(FailOnTimeout.java:74)
Caused by: java.io.IOException: forced error
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:561)
	... 11 more
2017-05-23 00:52:53,752 WARN  [Thread-10] reduce.Fetcher (Fetcher.java:openShuffleUrl(287)) - Failed to connect to localhost with 2 map outputs
java.io.IOException: Verification of the hashReply failed
	at org.apache.hadoop.mapreduce.security.SecureShuffleUtils.verifyReply(SecureShuffleUtils.java:105)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.verifyConnection(Fetcher.java:474)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.setupConnectionsWithRetry(Fetcher.java:407)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.openShuffleUrl(Fetcher.java:271)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:356)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostWithRetryUnreserve(TestFetcher.java:717)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$StatementThread.run(FailOnTimeout.java:74)
2017-05-23 00:52:53,754 INFO  [main] reduce.TestFetcher (TestFetcher.java:teardown(124)) - <<<< testCopyFromHostWithRetryUnreserve
2017-05-23 00:52:53,755 INFO  [main] reduce.TestFetcher (TestFetcher.java:setup(99)) - >>>> testInterruptOnDisk
2017-05-23 00:52:53,893 INFO  [fetcher#7] reduce.Fetcher (Fetcher.java:copyMapOutput(558)) - fetcher#7 about to shuffle output of map attempt_0_0001_m_000001_1 decomp: 10 len: 10 to DISK
2017-05-23 00:52:53,901 WARN  [fetcher#7] util.NativeCodeLoader (NativeCodeLoader.java:<clinit>(60)) - Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
2017-05-23 00:52:53,905 WARN  [fetcher#7] reduce.Fetcher (Fetcher.java:copyMapOutput(564)) - Failed to shuffle for fetcher#7
java.io.IOException: underlying stream closed, triggered an error
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher$StuckInputStream.freeze(TestFetcher.java:786)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher$StuckInputStream.read(TestFetcher.java:816)
	at java.io.DataInputStream.read(DataInputStream.java:149)
	at org.apache.hadoop.mapred.IFileInputStream.doRead(IFileInputStream.java:209)
	at org.apache.hadoop.mapred.IFileInputStream.readWithChecksum(IFileInputStream.java:189)
	at org.apache.hadoop.mapreduce.task.reduce.OnDiskMapOutput.doShuffle(OnDiskMapOutput.java:105)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:561)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:348)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.run(Fetcher.java:198)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher$FakeFetcher.run(TestFetcher.java:721)
2017-05-23 00:52:53,906 WARN  [fetcher#7] reduce.Fetcher (Fetcher.java:copyMapOutput(600)) - Failed to shuffle output of attempt_0_0001_m_000001_1 from localhost
java.io.IOException: java.io.IOException: underlying stream closed, triggered an error
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:565)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:348)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.run(Fetcher.java:198)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher$FakeFetcher.run(TestFetcher.java:721)
Caused by: java.io.IOException: underlying stream closed, triggered an error
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher$StuckInputStream.freeze(TestFetcher.java:786)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher$StuckInputStream.read(TestFetcher.java:816)
	at java.io.DataInputStream.read(DataInputStream.java:149)
	at org.apache.hadoop.mapred.IFileInputStream.doRead(IFileInputStream.java:209)
	at org.apache.hadoop.mapred.IFileInputStream.readWithChecksum(IFileInputStream.java:189)
	at org.apache.hadoop.mapreduce.task.reduce.OnDiskMapOutput.doShuffle(OnDiskMapOutput.java:105)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:561)
	... 3 more
2017-05-23 00:52:53,907 WARN  [fetcher#7] reduce.Fetcher (Fetcher.java:copyFromHost(364)) - copyMapOutput failed for tasks [attempt_0_0001_m_000001_1]
2017-05-23 00:52:53,909 INFO  [main] reduce.TestFetcher (TestFetcher.java:teardown(124)) - <<<< testInterruptOnDisk
2017-05-23 00:52:53,909 INFO  [main] reduce.TestFetcher (TestFetcher.java:setup(99)) - >>>> testCopyFromHostWait
2017-05-23 00:52:53,931 INFO  [main] reduce.Fetcher (Fetcher.java:copyMapOutput(548)) - fetcher#10 - MergeManager returned status WAIT ...
2017-05-23 00:52:53,933 INFO  [main] reduce.TestFetcher (TestFetcher.java:teardown(124)) - <<<< testCopyFromHostWait
2017-05-23 00:52:53,934 INFO  [main] reduce.TestFetcher (TestFetcher.java:setup(99)) - >>>> testInterruptInMemory
2017-05-23 00:52:53,962 INFO  [fetcher#2] reduce.Fetcher (Fetcher.java:copyMapOutput(558)) - fetcher#2 about to shuffle output of map attempt_0_0001_r_000001_1 decomp: 10 len: 10 to MEMORY
2017-05-23 00:52:54,025 WARN  [fetcher#2] reduce.Fetcher (Fetcher.java:copyMapOutput(564)) - Failed to shuffle for fetcher#2
java.io.IOException: underlying stream closed, triggered an error
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher$StuckInputStream.freeze(TestFetcher.java:786)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher$StuckInputStream.read(TestFetcher.java:816)
	at java.io.DataInputStream.read(DataInputStream.java:149)
	at org.apache.hadoop.mapred.IFileInputStream.doRead(IFileInputStream.java:209)
	at org.apache.hadoop.mapred.IFileInputStream.read(IFileInputStream.java:152)
	at org.apache.hadoop.io.IOUtils.readFully(IOUtils.java:207)
	at org.apache.hadoop.mapreduce.task.reduce.InMemoryMapOutput.doShuffle(InMemoryMapOutput.java:90)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:561)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:348)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.run(Fetcher.java:198)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher$FakeFetcher.run(TestFetcher.java:721)
2017-05-23 00:52:54,026 WARN  [fetcher#2] reduce.Fetcher (Fetcher.java:copyMapOutput(600)) - Failed to shuffle output of attempt_0_0001_m_000001_1 from localhost
java.io.IOException: java.io.IOException: underlying stream closed, triggered an error
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:565)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:348)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.run(Fetcher.java:198)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher$FakeFetcher.run(TestFetcher.java:721)
Caused by: java.io.IOException: underlying stream closed, triggered an error
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher$StuckInputStream.freeze(TestFetcher.java:786)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher$StuckInputStream.read(TestFetcher.java:816)
	at java.io.DataInputStream.read(DataInputStream.java:149)
	at org.apache.hadoop.mapred.IFileInputStream.doRead(IFileInputStream.java:209)
	at org.apache.hadoop.mapred.IFileInputStream.read(IFileInputStream.java:152)
	at org.apache.hadoop.io.IOUtils.readFully(IOUtils.java:207)
	at org.apache.hadoop.mapreduce.task.reduce.InMemoryMapOutput.doShuffle(InMemoryMapOutput.java:90)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:561)
	... 3 more
2017-05-23 00:52:54,026 WARN  [fetcher#2] reduce.Fetcher (Fetcher.java:copyFromHost(364)) - copyMapOutput failed for tasks [attempt_0_0001_m_000001_1]
2017-05-23 00:52:54,027 INFO  [main] reduce.TestFetcher (TestFetcher.java:teardown(124)) - <<<< testInterruptInMemory
2017-05-23 00:52:54,028 INFO  [main] reduce.TestFetcher (TestFetcher.java:setup(99)) - >>>> testCopyFromHostOnAnyException
2017-05-23 00:52:54,047 INFO  [Thread-17] reduce.Fetcher (Fetcher.java:copyMapOutput(558)) - fetcher#11 about to shuffle output of map null decomp: 10 len: 10 to null
2017-05-23 00:52:54,048 WARN  [Thread-17] reduce.Fetcher (Fetcher.java:copyMapOutput(564)) - Failed to shuffle for fetcher#11
java.lang.ArrayIndexOutOfBoundsException
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:561)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:348)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostOnAnyException(TestFetcher.java:398)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$StatementThread.run(FailOnTimeout.java:74)
2017-05-23 00:52:54,048 WARN  [Thread-17] reduce.Fetcher (Fetcher.java:copyMapOutput(600)) - Failed to shuffle output of attempt_0_0001_m_000001_1 from localhost
java.io.IOException: java.lang.ArrayIndexOutOfBoundsException
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:565)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:348)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostOnAnyException(TestFetcher.java:398)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$StatementThread.run(FailOnTimeout.java:74)
Caused by: java.lang.ArrayIndexOutOfBoundsException
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:561)
	... 11 more
2017-05-23 00:52:54,049 WARN  [Thread-17] reduce.Fetcher (Fetcher.java:copyFromHost(364)) - copyMapOutput failed for tasks [attempt_0_0001_m_000001_1]
2017-05-23 00:52:54,050 INFO  [main] reduce.TestFetcher (TestFetcher.java:teardown(124)) - <<<< testCopyFromHostOnAnyException
2017-05-23 00:52:54,051 INFO  [main] reduce.TestFetcher (TestFetcher.java:setup(99)) - >>>> testCopyFromHostWithRetryThenTimeout
2017-05-23 00:52:54,069 INFO  [Thread-19] reduce.Fetcher (Fetcher.java:copyMapOutput(558)) - fetcher#12 about to shuffle output of map null decomp: 10 len: 10 to null
2017-05-23 00:52:54,070 WARN  [Thread-19] reduce.Fetcher (Fetcher.java:copyMapOutput(564)) - Failed to shuffle for fetcher#12
java.io.IOException: forced error
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:561)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:348)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostWithRetryThenTimeout(TestFetcher.java:476)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$StatementThread.run(FailOnTimeout.java:74)
2017-05-23 00:52:54,070 WARN  [Thread-19] reduce.Fetcher (Fetcher.java:checkTimeoutOrRetry(622)) - Shuffle output from localhost failed, retry it.
java.io.IOException: java.io.IOException: forced error
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:565)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:348)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostWithRetryThenTimeout(TestFetcher.java:476)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$StatementThread.run(FailOnTimeout.java:74)
Caused by: java.io.IOException: forced error
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:561)
	... 11 more
2017-05-23 00:52:54,072 WARN  [Thread-19] reduce.Fetcher (Fetcher.java:openShuffleUrl(287)) - Failed to connect to localhost with 2 map outputs
java.net.SocketTimeoutException: forced timeout
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.verifyConnection(Fetcher.java:441)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.setupConnectionsWithRetry(Fetcher.java:407)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.openShuffleUrl(Fetcher.java:271)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:356)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostWithRetryThenTimeout(TestFetcher.java:476)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$StatementThread.run(FailOnTimeout.java:74)
2017-05-23 00:52:54,073 INFO  [main] reduce.TestFetcher (TestFetcher.java:teardown(124)) - <<<< testCopyFromHostWithRetryThenTimeout
2017-05-23 00:52:54,074 INFO  [main] reduce.TestFetcher (TestFetcher.java:setup(99)) - >>>> testCopyFromHostConnectionTimeout
2017-05-23 00:52:54,093 WARN  [Thread-21] reduce.Fetcher (Fetcher.java:openShuffleUrl(287)) - Failed to connect to localhost with 2 map outputs
java.io.IOException: Got invalid response code 0 from http://localhost:8080/attempt_0_0001_m_000001_1,attempt_0_0001_m_000002_1: null
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.verifyConnection(Fetcher.java:458)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.setupConnectionsWithRetry(Fetcher.java:407)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.openShuffleUrl(Fetcher.java:271)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:335)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostConnectionTimeout(TestFetcher.java:168)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$StatementThread.run(FailOnTimeout.java:74)
2017-05-23 00:52:54,105 INFO  [main] reduce.TestFetcher (TestFetcher.java:teardown(124)) - <<<< testCopyFromHostConnectionTimeout
2017-05-23 00:52:54,108 INFO  [main] reduce.TestFetcher (TestFetcher.java:setup(99)) - >>>> testReduceOutOfDiskSpace
2017-05-23 00:52:54,127 INFO  [main] reduce.TestFetcher (TestFetcher.java:testReduceOutOfDiskSpace(132)) - testReduceOutOfDiskSpace
2017-05-23 00:52:54,131 INFO  [main] reduce.TestFetcher (TestFetcher.java:teardown(124)) - <<<< testReduceOutOfDiskSpace
2017-05-23 00:52:54,132 INFO  [main] reduce.TestFetcher (TestFetcher.java:setup(99)) - >>>> testCopyFromHostExtraBytes
2017-05-23 00:52:54,155 INFO  [main] reduce.Fetcher (Fetcher.java:copyMapOutput(558)) - fetcher#15 about to shuffle output of map attempt_0_0001_m_000001_1 decomp: 10 len: 14 to MEMORY
2017-05-23 00:52:54,156 INFO  [main] reduce.InMemoryMapOutput (InMemoryMapOutput.java:doShuffle(93)) - Read 8 bytes from map-output for attempt_0_0001_m_000001_1
2017-05-23 00:52:54,156 WARN  [main] reduce.Fetcher (Fetcher.java:copyMapOutput(564)) - Failed to shuffle for fetcher#15
java.io.IOException: Unexpected extra bytes from input stream for attempt_0_0001_m_000001_1
	at org.apache.hadoop.mapreduce.task.reduce.InMemoryMapOutput.doShuffle(InMemoryMapOutput.java:105)
	at org.apache.hadoop.mapreduce.task.reduce.IFileWrappedMapOutput.shuffle(IFileWrappedMapOutput.java:63)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:561)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:348)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostExtraBytes(TestFetcher.java:521)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
2017-05-23 00:52:54,157 WARN  [main] reduce.Fetcher (Fetcher.java:copyMapOutput(600)) - Failed to shuffle output of attempt_0_0001_m_000001_1 from localhost
java.io.IOException: java.io.IOException: Unexpected extra bytes from input stream for attempt_0_0001_m_000001_1
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:565)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:348)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostExtraBytes(TestFetcher.java:521)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
Caused by: java.io.IOException: Unexpected extra bytes from input stream for attempt_0_0001_m_000001_1
	at org.apache.hadoop.mapreduce.task.reduce.InMemoryMapOutput.doShuffle(InMemoryMapOutput.java:105)
	at org.apache.hadoop.mapreduce.task.reduce.IFileWrappedMapOutput.shuffle(IFileWrappedMapOutput.java:63)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:561)
	... 29 more
2017-05-23 00:52:54,158 WARN  [main] reduce.Fetcher (Fetcher.java:copyFromHost(364)) - copyMapOutput failed for tasks [attempt_0_0001_m_000001_1]
2017-05-23 00:52:54,160 INFO  [main] reduce.TestFetcher (TestFetcher.java:teardown(124)) - <<<< testCopyFromHostExtraBytes
2017-05-23 00:52:54,161 INFO  [main] reduce.TestFetcher (TestFetcher.java:setup(99)) - >>>> testCorruptedIFile
2017-05-23 00:52:54,940 INFO  [main] reduce.OnDiskMapOutput (OnDiskMapOutput.java:doShuffle(117)) - Read 31 bytes from map-output for attempt_0_0001_m_000001_1
2017-05-23 00:52:54,941 INFO  [main] reduce.TestFetcher (TestFetcher.java:testCorruptedIFile(581)) - The expected checksum exception was thrown.
org.apache.hadoop.fs.ChecksumException: Checksum Error
	at org.apache.hadoop.mapred.IFileInputStream.doRead(IFileInputStream.java:228)
	at org.apache.hadoop.mapred.IFileInputStream.readWithChecksum(IFileInputStream.java:189)
	at org.apache.hadoop.mapreduce.task.reduce.OnDiskMapOutput.doShuffle(OnDiskMapOutput.java:105)
	at org.apache.hadoop.mapreduce.task.reduce.IFileWrappedMapOutput.shuffle(IFileWrappedMapOutput.java:63)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCorruptedIFile(TestFetcher.java:578)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
2017-05-23 00:52:54,949 INFO  [main] reduce.TestFetcher (TestFetcher.java:teardown(124)) - <<<< testCorruptedIFile
2017-05-23 00:52:54,954 INFO  [main] reduce.TestFetcher (TestFetcher.java:setup(99)) - >>>> testCopyFromHostBogusHeader
2017-05-23 00:52:54,993 WARN  [main] reduce.Fetcher (Fetcher.java:copyMapOutput(515)) - Invalid map id 
java.lang.IllegalArgumentException: TaskAttemptId string : 0 is not properly formed
	at org.apache.hadoop.mapreduce.TaskAttemptID.forName(TaskAttemptID.java:201)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:509)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:348)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostBogusHeader(TestFetcher.java:220)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
2017-05-23 00:52:54,994 WARN  [main] reduce.Fetcher (Fetcher.java:copyFromHost(364)) - copyMapOutput failed for tasks [attempt_0_0001_m_000002_1, attempt_0_0001_m_000001_1]
2017-05-23 00:52:54,997 INFO  [main] reduce.TestFetcher (TestFetcher.java:teardown(124)) - <<<< testCopyFromHostBogusHeader
2017-05-23 00:52:55,000 INFO  [main] reduce.TestFetcher (TestFetcher.java:setup(99)) - >>>> testCopyFromHostCompressFailure
2017-05-23 00:52:55,032 INFO  [Thread-30] reduce.Fetcher (Fetcher.java:copyMapOutput(558)) - fetcher#17 about to shuffle output of map null decomp: 10 len: 10 to null
2017-05-23 00:52:55,032 WARN  [Thread-30] reduce.Fetcher (Fetcher.java:copyMapOutput(564)) - Failed to shuffle for fetcher#17
java.lang.InternalError
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:561)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:348)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostCompressFailure(TestFetcher.java:361)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$StatementThread.run(FailOnTimeout.java:74)
2017-05-23 00:52:55,032 WARN  [Thread-30] reduce.Fetcher (Fetcher.java:copyMapOutput(600)) - Failed to shuffle output of attempt_0_0001_m_000001_1 from localhost
java.io.IOException: java.lang.InternalError
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:565)
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyFromHost(Fetcher.java:348)
	at org.apache.hadoop.mapreduce.task.reduce.TestFetcher.testCopyFromHostCompressFailure(TestFetcher.java:361)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$StatementThread.run(FailOnTimeout.java:74)
Caused by: java.lang.InternalError
	at org.apache.hadoop.mapreduce.task.reduce.Fetcher.copyMapOutput(Fetcher.java:561)
	... 11 more
2017-05-23 00:52:55,033 WARN  [Thread-30] reduce.Fetcher (Fetcher.java:copyFromHost(364)) - copyMapOutput failed for tasks [attempt_0_0001_m_000001_1]
2017-05-23 00:52:55,034 INFO  [main] reduce.TestFetcher (TestFetcher.java:teardown(124)) - <<<< testCopyFromHostCompressFailure
